tox.ini

```diff
@@ -12,3 +13,17 @@ deps =
   sphinx185: sphinx==1.8.5
   sphinx201: sphinx==2.0.1
 commands = pytest
+
+[testenv:py37-sphinx201]
+deps =
+  {[testenv]deps}
+  pytest-cov
+commands = pytest --cov=notfound tests/
+
+[testenv:docs]
+deps =
+  -r {toxinidir}/docs/requirements.txt
+  .
+changedir = {toxinidir}/docs
+commands =
+    sphinx-build -b html -d {envtmpdir}/doctrees . {envtmpdir}/html
```

Comment:
You could include `-W` to treat warnings as errors.

Reply:
Is there a way to turn this on and skip one specific warning?

There is going to be always a warning since `404.rst` is not included in any toctree.

Reply:
I'm not sure if this would break something, but sphinx has this option https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-exclude_patterns

And I don't think there is a way to remove specific warnings

Reply:
That kind of works. Adding this

    -W -D "exclude_patterns=404.rst" 

makes this to work. Although, any change done to `404.rst` file will be ignored when running this test.

I suppose that it's fine to merge as is and don't care too much about warnings for now, which is very strict for this project.